### DISCUSSION TITLE
Profiles for publishing

### DISCUSSION THREAD

**Discussion by gravesti** [quarto-dev/quarto-cli#4076] [Q&A] [answered]
Is it possible to use [project profiles](https://quarto.org/docs/projects/profiles.html) to change how projects are published?
eg, I would like to publish two versions of a project (`_quarto.yml` and `quarto-advanced.yml`) to 2 different RSConnect urls with `_publish.yml` and `_publish-advanced.yml`

**Comment by cderv** [comment]
Currently you can select a profile for the rendering in `quarto publish --profile` - See `quarto publish --help`

You can change the server in the command using `quarto publish --server` to select where. 

> Is it possible to use [project profiles](https://quarto.org/docs/projects/profiles.html) to change how projects are published?

Do you mean setting some publishing information into the `_quarto.yml` to modify with different file using project profiles feature ?

**Reply by andrewpbray** [reply]
@cderv 

I haven't been able to get this working. Here's a minimal repo: [https://github.com/andrewpbray/publish-profile](https://github.com/andrewpbray/publish-profile).

`quarto render --profile alternate` works just fine and `quarto publish` works just fine (it goes to `gh-pages`). But when I run `quarto publish --profile alternate` I get `ERROR: Publishing source 'alternate' not found`.

Is there something I'm missing with the syntax?

I have been able to publish profiles using environmental variables (see [this workflow](https://github.com/stat20/course-materials/blob/main/.github/workflows/publish-partial.yml)) but doing it via an option on `quarto publish` is where I get hung up.

**Reply by cderv** [reply]
Maybe we have a bug... I need to check - I would expect this to work as we document it in help page but maybe I am missing something

**Reply by cderv** [reply, marked as answer]
Yeah I think this is missing. Or either 

* Documented by not working
* Wrongly documented

As the env var is working I believe this is because it is not working 

Tracked in https://github.com/quarto-dev/quarto-cli/issues/4396

**Comment by gravesti** [comment]
Currently when I publish to RS Connect, a _publish.yml file is created:
```
- source: project
  connect:
    - id: 3b2b0000-0000-0000-0000-780a7eea5177
      url: 'https://connect.mycompany.com/content/3b2b0000-0000-0000-0000-780a7eea5177/'
```
This is reused when I publish again. 
I would like to have 2 of these configurations for different profiles. I thought it would be easy to have `_publish.yml` and `_publish-advanced.yml`.

Or can I put this into the `_quarto.yml`/`_quarto-advanced.yml` files?

**Reply by gravesti** [reply]
> Currently when I publish to RS Connect, a _publish.yml file is created:
> 
> - source: project
>   connect:
>     - id: 3b2b0000-0000-0000-0000-780a7eea5177
>       url: 'https://connect.mycompany.com/content/3b2b0000-0000-0000-0000-780a7eea5177/'
> This is reused when I publish again.
> I would like to have 2 of these configurations for different profiles. I thought it would be easy to have _publish.yml and _publish-advanced.yml.
> 
> Or can I put this into the _quarto.yml/_quarto-advanced.yml files?

@cderv Do you think it would ever be able to work like this? Or I should just set the id and url directly.

**Reply by cderv** [reply]
Current way of working is to pass id and url - you can have as many deployment in this file, and id is there to determine which one to use. 

Maybe it would be easier to add publishing information into profile... This needs to be thing through so maybe one day it will be possible. 

For now I would work with id.

**Comment by ejdataeditor** [comment]
hi all,
I'm still looking for a solution to this problem. could you please point me to the docs for env vars which explains how I can deploy to different gh-pages URLs? My use case is that I have on project which builds 2 slightly different websites. this works amazingly well with `--profile`. I just wish I could do `quarto publish gh-pages --profile xyz` as well. could you please explain how to set up the env var (which var?) would modify the remote URL in the `quarto publish gh-pages` call? thanks a lot

**Reply by mcanouil** [reply]
Set different `output-dir`, then it should work.

**Reply by mcanouil** [reply]
@ejdataeditor Note also that this thread is outdated, see the reply marked as answer.
